cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -242,10 +243,9 @@ public InetAddress providesHostname(CConfiguration cConf) {
     );
   }
 
-  private ProgramId getProgramIdFromRequest(ApplicationId preview, AppRequest request) {
-    // TODO remove this when UI passes the program name and type in the AppRequest.
+  private ProgramId getProgramIdFromRequest(ApplicationId preview, AppRequest request) throws BadRequestException {
     if (request.getPreview() == null) {
-      return preview.workflow("DataPipelineWorkflow");
+      throw new BadRequestException("preview config cannot be null");
```

Comment:
Though is that actually the case? The problem is that getPreview() is returning null; is the preview config the only source of that as the problem? (I know this is a cherry-pick, and perhaps this should go back to the original, but...)